sdk/nodejs/runtime/rpc.ts

```diff
@@ -419,6 +419,19 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
         return result;
     }
 
+    if (prop instanceof Map) {
+        const result: any = {};
+        for (const k of prop.keys()) {
+            if (excessiveDebugOutput) {
+                log.debug(`Serialize property [${ctx}]: map[${k}] element`);
+            }
+            // When serializing maps, we serialize any undefined values as `null`. This matches JSON semantics.
+            const elem = await serializeProperty(`${ctx}[${k}]`,prop.get(k), dependentResources);
+            result[k] = (elem === undefined ? null : elem);
```

Comment:
This doesn't look right. I just tried it out and calling `JSON.stringify` on a `Map` with a `undefined` value, and the entire key/value pair is excluded from the resulting JSON string.

```
$ node
Welcome to Node.js v14.16.1.
Type ".help" for more information.
> const m = new Map();
undefined
> m.set("foo", undefined);
Map(1) { 'foo' => undefined }
> JSON.stringify(m);
'{}'
```

If we decide we do want to add support for serializing `Map`, I think we want to behave the same as `serializeAllKeys` below.